src/analyzer/analyzersilence.cpp

```diff
@@ -117,29 +94,37 @@ void AnalyzerSilence::storeResults(TrackPointer tio) {
     double introStart = mixxx::kAnalysisChannels * m_iSignalStart;
     double outroEnd = mixxx::kAnalysisChannels * m_iSignalEnd;
 
-    if (shouldUpdateMainCue(tio->getCuePoint())) {
-        tio->setCuePoint(CuePosition(introStart, Cue::AUTOMATIC));
+    CuePointer pAudibleSound = pTrack->findCueByType(Cue::Type::AudibleSound);
+    if (pAudibleSound == nullptr) {
+        pAudibleSound = pTrack->createAndAddCue();
+        pAudibleSound->setType(Cue::Type::AudibleSound);
+        pAudibleSound->setPosition(introStart);
```

Comment:
I think we want to update the position and length in any case.

Reply:
What do you mean? The position and length are both set here.

Reply:
There can be the case that the track has changed and the user reanalyzes the track.
In this case we just need to update the position and length. The cue point itself already exists.

Reply:
Currently we have no chance to detect that automatically. Normally the user knows when he has edited the track and can manually reanalyze the track.
How can we archeve this here? By deleting intro and outro?

Reply:
Yes, I just implemented that in 0bc70f6 and 61e2e28.